### DISCUSSION TITLE
Mongo shell usage after upgrade from v2 to v4

### DISCUSSION THREAD

**Discussion by bendherville** [flapdoodle-oss/de.flapdoodle.embed.mongo#522] [Q&A]
Hi,
We are upgrading embed.mongo in our application as we have recently moved from windows to mac (m3) for development and the v2 version of embed.mongo doesn't work with arm64. Previously we programatically used mongo shell to run some scripts to setup our DB before our integration tests were executed.

I'm trying to do the same in v4.14.0 but I am getting an error when MongoShell is initialising.  
The code looks like;
```
    public void runScript(String script, ServerAddress serverAddress) throws IOException {
        Transitions transitions = MongoShell.builder().build().transitions(Version.V7_0_11)
                .replace(Start.to(MongoShellArguments.class).initializedWith(MongoShellArguments.builder().scriptName(script).build()))
                .addAll(Start.to(ServerAddress.class).initializedWith(serverAddress));

        try (TransitionWalker.ReachedState<ExecutedMongoShellProcess> state = transitions.walker()
                .initState(StateID.of(ExecutedMongoShellProcess.class))) {
            // TODO: check state
        }
    }
```
and the error is;
`java.lang.IllegalStateException: Cannot build ExtractedFileSet, some of required attributes are not set [executable]`

But digging into it a bit, it appears it's downloading an archive and looking for a file named '.*mongo' with the archive.  Presumably it's looking for the mongo shell command.

The archive it's downloaded is `/Users/bend/.embedmongo/archives/https-fastdl-mongodb-org/56cb3d10404e5b9df5358ac4ee83efcb/osxmongodb-macos-arm64-7-0-11-tgz/bdaa5f3450eea95884efe91ca458f071e85217364d4d13665376dc8c51e1d319/archive.tgz`

This contains mongod and mongos executables only.  

MongoShell is defaulting the Command to Mongo, which is where that .*mongo pattern is coming from.  In newer versions of mongodb the mongo shell command is called mongosh and distributed separate from the database package so I guess this is why it's not working but not entirely sure.

Can anyone confirm that MongoShell should work with mongo version v7_0 on MacOS?

**Comment by michaelmosmann** [comment]
@bendherville i will have a look.. it should download a matching tools version, because the tools are not part of the main release anymore..

**Comment by michaelmosmann** [comment]
@bendherville .. found the bug, will create a new release soon..

**Comment by michaelmosmann** [comment]
@bendherville .. release is ahead.. but with one surprise.. mongodb removed the mongo shell executable (mongo) with version >=6.x.x

I am not sure what kind of "setup our DB before our integration tests" you are using, but maybe there is an other way.

An example of using mongo shell with one work around if you need mongodb >=6.x.x: https://github.com/flapdoodle-oss/de.flapdoodle.embed.mongo/blob/main/docs/UseCases.md#execute-mongo-shell-with-running-mongod-server

**Reply by bendherville** [reply]
Thanks @michaelmosmann Yes, mongo was removed from the main mongodb distribution and moved into a separate package -- https://www.mongodb.com/try/download/shell 
The binary is now called `mongosh` instead of `mongo`
From my naive understanding of your project, it seems you could add another `Command` called Mongosh and with this download the package above and look for the mongosh executable.

**Reply by michaelmosmann** [reply]
@bendherville this might be possible.. but not without some changes.. so not a very fast solution for your problem..

**Reply by bendherville** [reply]
It's working with mongo 4.0 so we will live with that for now but we really should be on v7 as per our production environments.  
I think this issue would be present for mongo versions after 4.2.  Do you have an issue so I can track progress?

**Comment by michaelmosmann** [comment]
@bendherville we can create an issue from this:)

**Comment by michaelmosmann** [comment]
@bendherville thats the issue: https://github.com/flapdoodle-oss/de.flapdoodle.embed.mongo/issues/524

**Reply by benatportland** [reply]
Hi @michaelmosmann I see you assigned the issue, any updates?

**Reply by michaelmosmann** [reply]
@benatportland .. i did think about that a lot but did not make any progress.
